app/components/pay-page/choose-membership-plan-modal/plan-card.ts

```diff
@@ -0,0 +1,73 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import type TimeService from 'codecrafters-frontend/services/time';
+import { formatTimeDurationForCountdown } from 'codecrafters-frontend/utils/time-formatting';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type { PricingPlan } from '../choose-membership-plan-modal';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
+    isSelected: boolean;
+    onPlanSelect: (plan: PricingPlan) => void;
+    plan: PricingPlan;
+    regionalDiscount?: RegionalDiscountModel | null;
+  };
+}
+
+export default class PlanCard extends Component<Signature> {
+  @service declare time: TimeService;
+
+  get amortizedMonthlyPriceInDollars(): number | null {
+    if (this.args.plan.validityInMonths) {
+      return Math.round(this.effectivePriceInDollars / this.args.plan.validityInMonths);
+    }
+
+    return null;
+  }
+
+  get discountedPriceInDollars(): number | null {
+    const basePrice = this.args.plan.priceInDollars;
+    let discountedPrice = basePrice;
+
+    // Apply promotional discount first (yearly plans only)
+    if (this.args.activeDiscountForYearlyPlan && this.args.plan.id === 'yearly') {
+      discountedPrice = this.args.activeDiscountForYearlyPlan.computeDiscountedPrice(basePrice);
+    }
+
+    // Apply regional discount
+    if (this.args.regionalDiscount) {
+      const regionalDiscountAmount = (discountedPrice * this.args.regionalDiscount.percentOff) / 100;
```

Comment:
Let's add `computeDiscountedPrice` to regionalDiscount too, to be symmetric with `PromotionalDiscountModel`, that'll make this similar to the above line to read